File: java/word-count/src/main/java/WordCount.java
import java.util.HashMap;
import java.util.Map;

public class WordCount {

    public Map<String, Integer> phrase(String phrase) {
        String[] sanitizedPhrase = phrase.replaceAll("[^ \\w+]", "").toLowerCase().split("\\s+");
        HashMap<String, Integer> retVal = new HashMap<>();
        for(String word : sanitizedPhrase) {
            if(retVal.containsKey(word)) {
                retVal.put(word, retVal.get(word) + 1);
            } else {
                retVal.put(word, 1);
            }
        }
        return retVal;
    }
}

File: ruby/word-count/word_count.rb
class Phrase
  attr_reader :word_count
  VERSION = 1

  def initialize(phrase)
    @word_count = Hash.new(0)
    phrase.downcase.scan(/\w+'+\w+|\w+/).each do |word|
      @word_count[word] += 1
    end
    @word_count
  end
end
